tests/test_dimension.py

```diff
@@ -88,6 +89,24 @@ def test_subdim_middle(self):
         assert np.all(u.data[1, -1, :, :] == 1)
         assert np.all(u.data[1, 1:3, :, :] == 2)
 
+    @skipif('yask')
+    def test_symbolic_size(self):
+        """Check the symbolic size of all possible SubDimensions is as expected."""
+        grid = Grid(shape=(4,))
+        x, = grid.dimensions
+        thickness = 4
```

Comment:
does it make sense in trying to stick `thickness` with `grid.shape[0]` for safety ?

Reply:
they're identical, true, but it's just a coincidence, conceptually these are two different things